src/Client.php

```diff
@@ -92,49 +90,44 @@ class Client extends BaseClient
      *                                   string, e.g. "en-US" to fetch content in that locale. Set it to "*"
      *                                   to fetch content in all locales.
      * @param array       $options       An array of optional configuration options. The following keys are possible:
-     *                                   * guzzle      Override the guzzle instance used by the Contentful client
-     *                                   * logger      Inject a Contentful logger
-     *                                   * uriOverride Override the uri that is used to connect to the Contentful API (e.g. 'https://cdn.contentful.com/'). The trailing slash is required.
-     *                                   * cache       Null or a PSR-6 cache item pool. The client only writes to the cache if autoWarmup is true, otherwise, you are responsible for warming it up using \Contentful\Delivery\Cache\CacheWarmer.
-     *                                   * autoWarmup  Warm up the cache automatically
+     *                                   * guzzle     Override the guzzle instance used by the Contentful client
+     *                                   * logger     Inject a Contentful logger
+     *                                   * baseUri    Override the uri that is used to connect to the Contentful API (e.g. 'https://cdn.contentful.com/'). The trailing slash is required.
+     *                                   * cache      Null or a PSR-6 cache item pool. The client only writes to the cache if autoWarmup is true, otherwise, you are responsible for warming it up using \Contentful\Delivery\Cache\CacheWarmer.
+     *                                   * autoWarmup Warm up the cache automatically
      */
     public function __construct($token, $spaceId, $preview = false, $defaultLocale = null, array $options = [])
     {
-        $baseUri = $preview ? 'https://preview.contentful.com/' : 'https://cdn.contentful.com/';
-
         $options = \array_replace([
             'guzzle' => null,
             'logger' => null,
-            'uriOverride' => null,
+            'baseUri' => null,
             'cacheDir' => null,
             'cache' => null,
             'autoWarmup' => false,
         ], $options);
 
-        $guzzle = $options['guzzle'];
-        $logger = $options['logger'];
-        $uriOverride = $options['uriOverride'];
-        $this->autoWarmup = $options['autoWarmup'];
-
-        if (null !== $uriOverride) {
-            $baseUri = $uriOverride;
+        $baseUri = $preview ? self::URI_PREVIEW : self::URI_DELIVERY;
```

Comment:
This should be extracted to a method.

Reply:
I can think about it, but it'd be probably overkill...

Reply:
Not this particular line, I mean the whole snippet of `baseUri` mangling